src/library.js

```diff
@@ -824,12 +824,64 @@ LibraryManager.library = {
 #if SIMD
       196608
 #else
+#if WASM_ONLY
+      16384
+#else
       8192
 #endif
+#endif
     ) {
       return _emscripten_memcpy_big(dest|0, src|0, num|0)|0;
     }
 
+#if WASM_ONLY
+    ret = dest|0;
+    dest_end = (dest + num)|0;
+    if ((dest&7) == (src&7)) {
```

Comment:
In wasm-only mode, this raises an interesting question: if src or dst are unaligned, is it faster to do unaligned 8-byte copies or should one resort to aligned 1-byte copies. On x86 definitely unaligned 8-byte copies will still be faster than 1-byte copies, since x86 practically doesn't care (iirc it was a 1 cycle penalty either on loads or stores on Intel, and 1 cycle penalty on loads and stores on AMD). However on ARM this might be a different thing. Anyone have a super efficient ARM memcpy at hand?